src/data/moviesSlice.js

```diff
@@ -0,0 +1,27 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit"
+
+export const fetchMovies = createAsyncThunk('fetch-movies', async (apiUrl) => {
+    const response = await fetch(apiUrl)
+    return response.json()
```

Comment:
While the code fetches data, it might benefit from more comprehensive error handling to gracefully manage scenarios such as network failures, API errors, or other edge cases.